backbone.js

```diff
@@ -1511,13 +1535,18 @@
 
     // Convert a route string into a regular expression, suitable for matching
     // against the current location hash.
-    _routeToRegExp: function(route) {
+    _routeToRegExp: function(route, paramNames) {
```

Comment:
@akre54 
> I'm not so keen on modifying passed arguments. Let's bikeshed this decision a bit more (you may have been right in the first place).

I don't like this either, but this feels the least intrusive/most optimal since we are already RegExp parsing the input route string. And we can reuse `namedParam` and `splatParam` right where they are used.

Since `_routeToRegExp()` is only used from `route()`, how about moving it's body inside `route()` itself? Might be breaking change if someone is ovewritting `_routeToRegExp()` though.